src/Helper/Settings.php

```diff
@@ -26,77 +27,114 @@ final class Settings {
+  /**
+   * Get cache expiration.
+   */
+  public function getCacheExpiration(): string {
+    return $this->get(SettingsForm::CACHE_EXPIRATION, '');
+  }
+
+  /**
+   * Get organisation service endpoint.
+   */
+  public function getOrganisationServiceEndpoint(): string {
+    return $this->get(SettingsForm::ORGANISATION_SERVICE_ENDPOINT_REFERENCE, '');
   }
 
   /**
-   * {@inheritdoc}
+   * Get organisation test manager role id.
    */
-  public function getKeys(): array {
-    return $this->keys;
+  public function getOrganisationTestManagerRoleId(): string {
+    return $this->get(SettingsForm::ORGANISATION_TEST_LEDER_ROLLE_UUID, '');
   }
 
   /**
-   * {@inheritdoc}
+   * Get organisation production manager role id.
    */
-  public function get(string $key, $default = NULL) {
-    return $this->store->get($this->buildKey($key), $default);
+  public function getOrganisationProductionManagerRoleId(): string {
+    return $this->get(SettingsForm::ORGANISATION_PROD_LEDER_ROLLE_UUID, '');
   }
 
   /**
-   * {@inheritdoc}
+   * Get a setting value.
+   *
+   * @param string $key
+   *   The key.
+   * @param mixed|null $default
+   *   The default value.
+   *
+   * @return mixed
+   *   The setting value.
    */
-  public function set(string $key, $value): void {
-    $this->store->set($this->buildKey($key), $value);
+  private function get(string $key, $default = NULL) {
+    $resolver = $this->getSettingsResolver();
+    if (!$resolver->isDefined($key)) {
+      throw new InvalidSettingException(sprintf('Setting %s is not defined', $key));
+    }
+
+    return $this->store->get($key, $default);
   }
 
   /**
-   * Build key.
+   * Set settings.
+   *
+   * @throws \Symfony\Component\OptionsResolver\Exception\ExceptionInterface
+   *
+   * @phpstan-param array<string, mixed> $settings
    */
-  private function buildKey(string $key): string {
-    if (!in_array($key, $this->keys, TRUE)) {
-      throw new InvalidSettingException(sprintf('Invalid setting: %s', $key));
+  public function setSettings(array $settings): self {
+    $settings = $this->getSettingsResolver()->resolve($settings);
+    foreach ($settings as $key => $value) {
+      $this->store->set($key, $value);
     }
-    return $this->collection . $key;
+
+    return $this;
   }
 
   /**
-   * Unbuild key.
+   * Get settings resolver.
    */
-  private function unbuildKey(string $key): string {
-    return 0 === strpos($key, $this->collection) ? substr($key, strlen($this->collection)) : $key;
+  private function getSettingsResolver(): OptionsResolver {
+    return (new OptionsResolver())
+      ->setDefaults([
+        SettingsForm::TEST_MODE => TRUE,
+        SettingsForm::AUTHORITY_CVR => '',
+        SettingsForm::CERTIFICATE => [],
+        SettingsForm::CACHE_EXPIRATION => '',
+        SettingsForm::ORGANISATION_SERVICE_ENDPOINT_REFERENCE => '',
+        SettingsForm::ORGANISATION_TEST_LEDER_ROLLE_UUID => '',
+        SettingsForm::ORGANISATION_PROD_LEDER_ROLLE_UUID => '',
```

Comment:
It's a little weird that these constants reside in `SettingsForm` (and this gives us a mutual dependency between the two services). It may make more sende to put them in `Settings`.